library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -533,13 +588,14 @@ class Client() {
     suspend fun publish(envelopes: List<Envelope>) {
         val authorized = AuthorizedIdentity(
             address = address,
-            authorized = privateKeyBundleV1.identityKey.publicKey,
-            identity = privateKeyBundleV1.identityKey,
+            authorized = v1keys.identityKey.publicKey,
+            identity = v1keys.identityKey,
         )
+        val client = apiClient ?: throw XMTPException("V2 only function")
```

Comment:
just a nit, we could check for null `apiClient` first in the function, since there's no need creating AuthorizedIdentity if apiClient is null.